chain-core/src/fixed/div.rs

```diff
@@ -0,0 +1,114 @@
+use core::ops;
+
+use super::{hi_lo, neg_abs, Fixed, FRAC_NBITS, NBITS};
+
+impl ops::Div for Fixed {
+    type Output = Fixed;
+
+    #[inline]
+    fn div(self: Fixed, other: Fixed) -> Fixed {
+        let (ans, overflow) = div_overflow(self.0, other.0, FRAC_NBITS);
+        debug_assert!(!overflow, "overflow");
+        Fixed(ans)
+    }
+}
+
+#[inline]
+fn div_overflow(lhs: i128, rhs: i128, frac_nbits: u32) -> (i128, bool) {
+    if frac_nbits == 0 {
+        lhs.overflowing_div(rhs)
+    } else {
+        let lhs2 = (lhs >> (NBITS - frac_nbits), (lhs << frac_nbits) as u128);
+
+        let (d_neg, d_abs) = neg_abs(rhs);
+        let (n_neg, n_abs) = neg_abs2(lhs2);
+        let (q, _) = div_rem_from(d_abs, n_abs);
+        let quot2 = from_neg_abs(n_neg != d_neg, q);
+
+        let quot = quot2.1 as i128;
+        let overflow = quot2.0 != if quot < 0 { -1 } else { 0 };
+        (quot, overflow)
+    }
+}
+
+#[inline]
+fn div_rem_from(dividor: u128, dividend: (u128, u128)) -> ((u128, u128), u128) {
+    let (mut n1, mut n0, mut d) = (dividend.0, dividend.1, dividor);
+    let (mut r, zeros) = normalize(&mut d, &mut n1, &mut n0);
+
+    let (n1_hi, n1_lo) = hi_lo(n1);
+    let q1h = div_half(&mut r, d, n1_hi);
+    let q1l = div_half(&mut r, d, n1_lo);
+    let (n0_hi, n0_lo) = hi_lo(n0);
+    let q0h = div_half(&mut r, d, n0_hi);
+    let q0l = div_half(&mut r, d, n0_lo);
+    ((up_lo(q1h, q1l), up_lo(q0h, q0l)), unnormalize(r, zeros))
+}
+
+#[inline]
+fn neg_abs2(num: (i128, u128)) -> (bool, (u128, u128)) {
```

Comment:
Some documentation on what this function does?

Reply:
Done